File: web/src/types/ip-group.ts
/**
 * IP组数据结构
 */
export interface IPGroup {
    id: string             // IP组唯一标识符
    name: string           // IP组名称
    items: string[]        // IP地址或CIDR列表
}

/**
 * IP组创建请求
 */
export interface IPGroupCreateRequest {
    name: string           // IP组名称
    items: string[]        // IP地址或CIDR列表
}

/**
 * IP组更新请求
 */
export interface IPGroupUpdateRequest {
    name?: string          // IP组名称
    items?: string[]       // IP地址或CIDR列表
}

/**
 * IP组列表响应
 */
export interface IPGroupListResponse {
    total: number          // 总数
    items: IPGroup[]       // IP组列表
}

/**
 * 用于表单的IP组数据
 */
export interface IPGroupFormValues {
    name: string
    items: string[]
}

File: server/dto/ip_group.go
// server/dto/ip_group.go
package dto

import (
	"github.com/HUAHUAI23/RuiQi/pkg/model"
)

// IPGroupCreateRequest IP组创建请求
// @Description 创建IP组的请求参数
type IPGroupCreateRequest struct {
	Name  string   `json:"name" binding:"required" example:"内部服务器"`              // IP组名称
	Items []string `json:"items" binding:"required" example:"[\"192.168.1.1\"]"` // IP地址或CIDR列表
}

// IPGroupUpdateRequest IP组更新请求
// @Description 更新IP组的请求参数
type IPGroupUpdateRequest struct {
	Name  string   `json:"name,omitempty" example:"内部服务器"`              // IP组名称
	Items []string `json:"items,omitempty" example:"[\"192.168.1.1\"]"` // IP地址或CIDR列表
}

// IPGroupListResponse IP组列表响应
// @Description IP组列表响应
type IPGroupListResponse struct {
	Total int64           `json:"total"` // 总数
	Items []model.IPGroup `json:"items"` // IP组列表
}

// AddIPToBlacklistRequest 添加IP到黑名单请求
// @Description 添加IP地址或CIDR到系统默认黑名单的请求
type AddIPToBlacklistRequest struct {
	IP string `json:"ip" binding:"required" example:"192.168.1.1"` // IP地址或CIDR
}
